src/halmos/processes.py

```diff
@@ -228,6 +250,25 @@ def _join(self):
                 future.result()
 
 
+# Global PopenExecutor instance for shared use across all tests and probes
+_global_executor: PopenExecutor | None = None
+_global_executor_lock = threading.Lock()
+
+
+def get_global_executor() -> PopenExecutor:
+    """Get the global PopenExecutor instance, creating it if necessary."""
+    global _global_executor
+    
+    if _global_executor is None:
+        with _global_executor_lock:
+            if _global_executor is None:
+                _global_executor = PopenExecutor()
+                # Register with the ExecutorRegistry so it gets shut down properly
+                ExecutorRegistry().register(_global_executor)
+    
+    return _global_executor
```

Comment:
the lock seems unnecessary. 

Wouldn't this work?

```
_executor = PopenExecutor()
ExecutorRegistry().register(_executor)

def get_global_executor() -> PopenExecutor:
    return _executor
```

We don't need to create the executor lazily, and instead could rely on the module initialization system to get it initialized once